### ISSUE TITLE
Find a foolproof way to validate URL on metadata

### ISSUE THREAD

**Issue by sumandari** [qgis/QGIS-Django#181]
We need to add a method to check the URL in metadata, we don't want URLs like `https://example.com` or `http://foo.bar.com` are accepted as a valid URL.

**Comment by sumandari** [comment]
I have put in a feature request to change the URL value to text as placeholder on the plugin builder at https://github.com/g-sherman/Qgis-Plugin-Builder/issues/133

**Comment by NyakudyaA** [comment]
> I have put in a feature request to change the URL value to text as placeholder on the plugin builder at [g-sherman/Qgis-Plugin-Builder#133](https://github.com/g-sherman/Qgis-Plugin-Builder/issues/133)

Maybe https://github.com/qgis/QGIS-Enhancement-Proposals/issues/219 will take care of this issue. @Gustry Is this in your road plan to fix URL validation in metadata.txt

**Comment by NyakudyaA** [comment]
> I have put in a feature request to change the URL value to text as placeholder on the plugin builder at [g-sherman/Qgis-Plugin-Builder#133](https://github.com/g-sherman/Qgis-Plugin-Builder/issues/133)

There seem to be other plugin builders out there as described here https://github.com/qgis/QGIS-Enhancement-Proposals/issues/223.

**Comment by Gustry** [comment]
What is not correct with `https://example.com` or `http://foo.bar.com`  ?

**Comment by sumandari** [comment]
I think we want to have something like github.com or gitlab.com for the repository and bug tracker instead of those URL, and also have a real homepage URL address.

**Comment by NyakudyaA** [comment]
> What is not correct with `https://example.com` or `http://foo.bar.com` ?

![url-s](https://user-images.githubusercontent.com/2510900/125045276-1adb5080-e09d-11eb-8613-6f3ec4154df9.png)

Although these URLs might be valid we are always 90% certain that those are not valid URL for the code repository as depicted above

**Comment by Gustry** [comment]
Ah ok sorry, you mean "these precise" URLs, not the URL scheme.

In the code, the easiest to blocklist "https://example.com"

**Comment by sumandari** [comment]
Agreed, we can create a list to block URLs. Is there any reference/link to get the public list of this kind of URL on the internet? So that we just can copy it to our list.

**Comment by Gustry** [comment]
I don't think we can't avoid a human check here.
We can only block URLs which are provided by default in tools generating plugins. So I dunderstood the one from Plugin builder is "example.com" ?

https://github.com/wonder-sk/qgis-minimal-plugin (which I use), doesn't use a valid URL, so ok
